src/consumer/stream_consumer.rs

```diff
@@ -89,35 +88,37 @@ impl<'a, C: ConsumerContext + 'a> Stream for MessageStream<'a, C> {
 
 /// Internal consumer loop. This is the main body of the thread that will drive the stream consumer.
 /// If `send_none` is true, the loop will send a None into the sender every time the poll times out.
-fn poll_loop<C: ConsumerContext>(
-    consumer: &BaseConsumer<C>,
+async fn poll_loop<C: ConsumerContext>(
+    consumer: Arc<BaseConsumer<C>>,
     mut sender: mpsc::Sender<Option<PolledMessagePtr>>,
-    should_stop: &AtomicBool,
+    should_stop: Arc<AtomicBool>,
     poll_interval: Duration,
     send_none: bool,
 ) {
-    trace!("Polling thread loop started");
+    trace!("Polling task loop started");
+    let _on_drop = OnDrop(|| trace!("Polling task loop terminated"));
     while !should_stop.load(Ordering::Relaxed) {
-        trace!("Polling base consumer");
-        let future_sender = match consumer.poll_raw(Timeout::After(poll_interval)) {
+        let backoff = time::delay_for(poll_interval);
```

Comment:
this strictly speaking is not poll interval and changes semantic of this value. Previously if poll_interval was 1minuted, but data arrived in 1 second, it were to become available in 1 second,  now it is 1 minutes.  This probably not a problem in practice, but it should at least be reflected in docs.

Reply:
actually it is a problem for low latency use cases.

Reply:
Yeah it's never really been the poll interval; it's always been the maximum amount of time to wait for a message. Your point about the low latency use cases is well taken though. I'll address that in a follow up PR.

Reply:
What was the rush to merge it? You knowingly introduced regression, I don't get it.

Reply:
This PR was already rather large. There are no stability guarantees on master. The problem will be fixed before the next release.